tests/baselines/reference/typeGuardConstructorClassAndNumber.errors.txt

```diff
@@ -0,0 +1,115 @@
+tests/cases/compiler/typeGuardConstructorClassAndNumber.ts(42,10): error TS2339: Property 'property1' does not exist on type 'number | C1'.
+  Property 'property1' does not exist on type 'number'.
+tests/cases/compiler/typeGuardConstructorClassAndNumber.ts(46,10): error TS2339: Property 'property1' does not exist on type 'number | C1'.
+  Property 'property1' does not exist on type 'number'.
+tests/cases/compiler/typeGuardConstructorClassAndNumber.ts(50,10): error TS2339: Property 'property1' does not exist on type 'number | C1'.
+  Property 'property1' does not exist on type 'number'.
+tests/cases/compiler/typeGuardConstructorClassAndNumber.ts(54,10): error TS2339: Property 'property1' does not exist on type 'number | C1'.
+  Property 'property1' does not exist on type 'number'.
+tests/cases/compiler/typeGuardConstructorClassAndNumber.ts(58,10): error TS2339: Property 'property1' does not exist on type 'number | C1'.
+  Property 'property1' does not exist on type 'number'.
+tests/cases/compiler/typeGuardConstructorClassAndNumber.ts(62,10): error TS2339: Property 'property1' does not exist on type 'number | C1'.
+  Property 'property1' does not exist on type 'number'.
+tests/cases/compiler/typeGuardConstructorClassAndNumber.ts(66,10): error TS2339: Property 'property1' does not exist on type 'number | C1'.
+  Property 'property1' does not exist on type 'number'.
+tests/cases/compiler/typeGuardConstructorClassAndNumber.ts(70,10): error TS2339: Property 'property1' does not exist on type 'number | C1'.
+  Property 'property1' does not exist on type 'number'.
+
+
+==== tests/cases/compiler/typeGuardConstructorClassAndNumber.ts (8 errors) ====
+    // Typical case
+    class C1 {
+        property1: string;
+    }
+    
+    let var1: C1 | number;
+    if (var1.constructor == C1) {
+        var1; // C1
+        var1.property1; // string
+    }
+    if (var1["constructor"] == C1) {
+        var1; // C1
+        var1.property1; // string
+    }
+    if (var1.constructor === C1) {
+        var1; // C1
+        var1.property1; // string
+    }
+    if (var1["constructor"] === C1) {
+        var1; // C1
+        var1.property1; // string
+    }
+    if (C1 == var1.constructor) {
+        var1; // C1
+        var1.property1; // string
+    }
+    if (C1 == var1["constructor"]) {
+        var1; // C1
+        var1.property1; // string
+    }
+    if (C1 === var1.constructor) {
+        var1; // C1
+        var1.property1; // string
+    }
+    if (C1 === var1["constructor"]) {
+        var1; // C1
+        var1.property1; // string
+    }
+    
+    if (var1.constructor != C1) {
+        var1; // C1 | number
+        var1.property1; // error
+             ~~~~~~~~~
+!!! error TS2339: Property 'property1' does not exist on type 'number | C1'.
+!!! error TS2339:   Property 'property1' does not exist on type 'number'.
+    }

```

Comment:
After you do my first comment's change, adding `else` branches to all these is pertinent.

Reply:
Will do 👍